api/src/room_store/merged_room_store.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+import logging
+from typing import (
+    List,
+    AsyncGenerator,
+    AsyncIterator,
+    Iterable,
+    Any,
+)
+
+from src.api.api_structures import Request, Action
+from src.room_store.room_archive import RoomArchive
+from src.room_store.room_store import (
+    RoomStore,
+    ReplacementData,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class MergedRoomStore(RoomStore):
+    def __init__(
+        self,
+        room_store: RoomStore,
+        room_archive: RoomArchive,
+    ):
+        self._room_store = room_store
+        self._room_archive = room_archive
+
+    async def changes(self, room_id: str) -> AsyncIterator[Request]:
+        return await self._room_store.changes(room_id)
+
+    async def get_all_room_ids(self) -> AsyncGenerator[str, None]:
+        async for room_id in self._room_store.get_all_room_ids():
+            yield room_id
+
+        async for room_id in self._room_archive.get_all_room_ids():
+            yield room_id
+
+    async def room_exists(self, room_id: str) -> bool:
+        return bool(
+            await self._room_store.room_exists(room_id)
+        ) or await self._room_archive.room_exists(room_id)
+
+    async def _load_into_redis(self, room_id: str) -> None:
+        if not await self._room_store.room_exists(
+            room_id
+        ) and await self._room_archive.room_exists(room_id):
```

Comment:
```suggestion
        if not await self.room_exists(room_id):
```

Reply:
This statement is not the same as room_exists